data_model/src/isi.rs

```diff
@@ -19,6 +18,9 @@ pub use self::model::*;
 use super::{expression::EvaluatesTo, prelude::*, IdBox, RegistrableBox, Value};
 use crate::Registered;
 
+/// Marker trait designating instruction
+pub trait Instruction {}
```

Comment:
> Should also be sealed I believe

no, it should be used in `client`. Compare `request` and `submit` methods. One is generic and the other is not

Reply:
*Sealed* means that it can't be implemented on any structure outside of this module. I don't suggest to make it private, I suggest doing this:

```rust
pub mod marker {
    mod sealed {
        pub trait Sealed {}

        impl Sealed for Instruction1 {}
        // ...
    }

    /// Marker trait designating instruction
    pub trait Instruction: Sealed {}

    impl Instruction for Instruction1 {}
    // ...
}
```

Reply:
yeah, my bad. I guess you are correct here

Reply:
done